Form 26-455 flaky test fix

bkjohnson opened the pull request:
## Description

This is fixing a flaky test caused by #19591 . It removes the usage of an undefined schema definition.

In the [source of the schema's definitions](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/c334f5c0d3233696aa00f7198590359b6e16e250/src/schemas/26-4555/schema.js#L6), the `bankAccount` schema is not defined. This means that there is nothing for the app code to import, which resulted in a flaky test depending on ordering:

![image](https://user-images.githubusercontent.com/2008881/145491267-58f18ec6-ff09-4123-a969-5d0c9b173a24.png)

## Testing done

`CHOMA_SEED=t8JQ4aC579 yarn test:unit`

## Screenshots


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

bkjohnson committed:
Add form to constants file

bkjohnson committed:
Rename

bkjohnson committed:
Use definitions from schema

bkjohnson committed:
Remove undefined schemas

bkjohnson commented on src/applications/sah/sahg/config/form.js:
```diff
@@ -1,31 +1,18 @@
-// In a real app this would not be imported directly; instead the schema you
-// imported above would import and use these common definitions:
-import commonDefinitions from 'vets-json-schema/dist/definitions.json';
+import fullSchema from 'vets-json-schema/dist/26-4555-schema.json';
```
Using the proper schema from the dependency.

bkjohnson commented on src/applications/sah/sahg/config/form.js:
```diff
@@ -35,9 +22,9 @@ const {
+  // bankAccount,
   toursOfDuty,
-} = commonDefinitions;
+} = fullSchema.definitions;
```
The schema for 26-4555 doesn't have a definition for `bankAccount`. This is the cause of the flaky test.

![image](https://user-images.githubusercontent.com/2008881/145492031-bd0e2e72-0ba2-4774-9238-50f5d0b25ec2.png)

https://github.com/department-of-veterans-affairs/vets-json-schema/blob/5b4775402bec413f6c55ee7288cdd05b1c048d37/dist/26-4555-schema.json#L6-L215

bkjohnson commented on src/applications/sah/sahg/config/form.js:
```diff
@@ -77,7 +64,7 @@ const formConfig = {
   introduction: IntroductionPage,
   confirmation: ConfirmationPage,
-  formId: '26-4555',
+  formId: VA_FORM_IDS.FORM_26_4555,
```
This is good hygiene.

bkjohnson reviewed (commented):
Tagging @jerekshoe for awareness.

jerekshoe commented:
Is it safe to use the generator still / Does the generator need to be updated to include these changes? I didn’t make any changes to the output from the generator

bkjohnson commented:
I'm not sure what generator you're referring to, but you'll probably need to do [something like this](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/328e2b65617b2caaa4f6535d1f1202dc7b1a3e1d/src/schemas/VRRAP/schema.js#L10). This is based on the ["create a new schema" documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/frontend/vets-website/creating-form-schema.md)

**[Schema tips](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/frontend/vets-website/creating-form-schema.md#schema-tips)**

> Use the common definitions as much as possible.

bkjohnson committed:
Remove local jsonschema

The one from the dependency exists for both `vets-website` and
`vets-api` to use. Relying on a schema specific to `vets-website` allows
version skew to happen, so a change to the frontend might break the API.

jerekshoe commented:
I just used the generator from https://github.com/department-of-veterans-affairs/generator-vets-website. I didn’t modify the files it created

bkjohnson commented:
It looks like you used the [complex](https://github.com/department-of-veterans-affairs/generator-vets-website/blob/9f149acbb856e4474336eb55251c62e0be3c5955/generators/form/templates/complex-form-schema.json.ejs#L6) option. Maybe the generator could be updated, though I'm not sure where we want to draw the line. My understanding is that the generator is meant to provide a starting template, and then teams can further customize it to their needs. We could have it automatically import all the common definitions, but I don't think every form app uses `bankAccount`, for example.

jerekshoe commented:
Sure it makes sense that it’s only a starter template, it’s just surprising that it seems to be broken out of the box. This is my first form so I appreciate you fixing the issues, it’s helpful to see what a good form is supposed to look like

bkjohnson commented:
To be clear, this "fix" is just for the flaky test. If you want this form to have the `bankAccount` definition, it will need to be added to the schema in `vets-json-schema`.

jerekshoe commented:
Ok cool, I’ll keep that in mind

k80bowman commented:
Thank you for digging into this @bkjohnson, this works well. Did you intend to leave so much code commented out?

bkjohnson commented:
@k80bowman yeah, I left the `bankAccount` pieces commented out so that there would still be a clear path on what to add back in and where it should go.
